utils/load_path.ml

```diff
@@ -96,8 +96,22 @@ let add dir =
     dir.Dir.files;
   dirs := dir :: !dirs
 
+let append_dir = add
+
 let add_dir dir = add (Dir.create dir)
 
+(* Add the directory at the start of load path - so basenames are
+   unconditionally added. *)
+let prepend_dir dir =
+  assert (not Config.merlin || Local_store.is_bound ());
+  List.iter
+    (fun base ->
+       let fn = Filename.concat dir.Dir.path base in
+       STbl.replace !files base fn;
+       STbl.replace !files_uncap (String.uncapitalize_ascii base) fn)
+    dir.Dir.files;
+  dirs := dir :: !dirs
```

Comment:
Shouldn't `dir` be added to the end of `dirs`? Otherwise theoretically one could get the wrong result when looking up non-basename files (see `find`).

Reply:
Thanks for catching that - that appears to be a rebase/committing artefact!